### PR TITLE
[JIT] Change SchemaCheckTensor into SchemaCheckMode and fix global variable issues

### PR DESCRIPTION
**Pull request by ttanpcs**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79474
* __->__ #79473
* #79316

### CODE REVIEW

File: torch/testing/_internal/schema_check_mode.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    def reset_cache(self):
+        self.ops.clear()
+
+    def display_ops(self):
+        print(*self.ops, sep=",")
+
+    def __torch_dispatch__(self, func, types, args=(), kwargs=None):
+        def has_mutated(before, after):
+            return not torch.equal(before, after) if isinstance(before, torch.Tensor) and isinstance(after, torch.Tensor) else False
```
Comment (eellison) [inline torch/testing/_internal/schema_check_mode.py:24]: Can we deepcopy before ? (unrelated to refactor)

**Review by eellison — APPROVED** [review]
LGTM

File: torch/testing/_internal/schema_check_mode.py (hunk 1 above)
Comment (goldenxuett) [reply torch/testing/_internal/schema_check_mode.py:24]: Yes we deepcopy before. (Although I'm not entirely sure if that's what you are asking here)
